src/database/migrations/20200924034242-student-overrides-setup.ts

```diff
@@ -0,0 +1,135 @@
+import { QueryInterface, DataTypes } from 'sequelize';
+
+export default {
+  up: async (queryInterface: QueryInterface): Promise<void> => {
+    // Transactions are automatically use because a namespace is injected into sequelize when fetching configurations
+    await queryInterface.sequelize.transaction(async () => {
+      /**
+       * student_topic_override
+       */
+      await queryInterface.createTable('student_topic_override', {
+        id: {
+          field: 'student_topic_override_id',
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true,
+        },
+        courseTopicContentId: {
+          field: 'course_topic_content_id',
+          type: DataTypes.INTEGER,
+          allowNull: false,
+          // Defaults to cascade when generating the db, no action on migrations
+          onUpdate: 'CASCADE',
+          // Different from sequelize model, on models I do this with associations so I can use types
+          references: {
+            model: 'course_topic_content',
+            key: 'course_topic_content_id',
+          },
+        },
+        userId: {
+          field: 'user_id',
+          type: DataTypes.INTEGER,
+          allowNull: false,
+          // Defaults to cascade when generating the db, no action on migrations
+          onUpdate: 'CASCADE',
+          // Different from sequelize model, on models I do this with associations so I can use types
+          references: {
+            model: 'users',
+            key: 'user_id',
+          },
+        },
+        endDate: {
+          field: 'student_topic_override_end_date',
+          type: DataTypes.DATE,
+          allowNull: true,
+        },
+        deadDate: {
+          field: 'student_topic_override_dead_date',
+          type: DataTypes.DATE,
+          allowNull: true,
+        },
```

Comment:
while we're at it, I suppose we might as well let profs override the startDate as well? I think that may be useful for a makeup exam, perhaps?

Reply:
I think that makes sense, if you specifically don't want a student to start something early.

Reply:
🙄 now you tell me 😆

Reply:
Resolved ^